Source/WebKit/UIProcess/API/glib/WebKitUserContentManager.cpp

```diff
@@ -224,12 +256,158 @@ void webkit_user_content_manager_remove_all_scripts(WebKitUserContentManager* ma
+    int referenceCount;
+};
+
+G_DEFINE_BOXED_TYPE(WebKitUserScriptMessageReply, webkit_user_script_message_reply, webkit_user_script_message_reply_ref, webkit_user_script_message_reply_unref)
+
+/**
+ * webkit_user_script_message_reply_ref:
+ * @user_script_message_reply*: A #WebKitUserScriptMessageReply
+ *
+ * Atomically increments the reference count of @user_script_message_reply by one.
+ *
+ * Returns: the @user_script_message_reply passed in.
+ *
+ * Since: 2.40
+ */
+WebKitUserScriptMessageReply*
+webkit_user_script_message_reply_ref(WebKitUserScriptMessageReply* userScriptMessageReply)
+{
+    g_return_val_if_fail(userScriptMessageReply, nullptr);
+    g_atomic_int_inc(&userScriptMessageReply->referenceCount);
+    return userScriptMessageReply;
+}
+
+/**
+ * webkit_user_script_message_reply_unref:
+ * @user_script_message_reply*: A #WebKitUserScriptMessageReply
+ *
+ * Atomically decrements the reference count of @user_script_message_reply by one.
+ *
+ * If the reference count drops to 0, all the memory allocated by the
+ * #WebKitUserScriptMessageReply is released. This function is MT-safe and may
+ * be called from any thread.
+ *
+ * Since: 2.40
+ */
+void webkit_user_script_message_reply_unref(WebKitUserScriptMessageReply* userScriptMessageReply)
+{
+    g_return_if_fail(userScriptMessageReply);
+    if (g_atomic_int_dec_and_test(&userScriptMessageReply->referenceCount)) {
+        userScriptMessageReply->~WebKitUserScriptMessageReply();
+        fastFree(userScriptMessageReply);
+    }
+}
+
+WebKitUserScriptMessageReply* webKitUserScriptMessageReplyCreate(WebCore::SerializedScriptValue& serializedScriptValue, WTF::Function<void(API::SerializedScriptValue*, const String&)>&& completionHandler)
+{
+    WebKitUserScriptMessageReply* userScriptMessageReply = static_cast<WebKitUserScriptMessageReply*>(fastMalloc(sizeof(WebKitUserScriptMessageReply)));
+    new (userScriptMessageReply) WebKitUserScriptMessageReply(serializedScriptValue, WTFMove(completionHandler));
+    return userScriptMessageReply;
+}
+
+/**
+ * webkit_user_script_message_reply_get_value:
+ * @user_script_message_reply: A #WebKitUserScriptMessageReply
+ *
+ * Return the value of reply sent in by user.
+ *
+ * Returns: (transfer none): The #JSCValue sent by JavaScript code to the
+ * script message handler.
+ *
+ * Since: 2.40
+ */
+JSCValue* webkit_user_script_message_reply_get_value(WebKitUserScriptMessageReply* userScriptMessageReply)
+{
+    return userScriptMessageReply->message.get();
+}
+
+/**
+ * webkit_user_script_message_reply_return_value:
+ * @user_script_message_reply: A #WebKitUserScriptMessageReply
+ * @reply_value: Reply value of the provided script message
+ *
+ * Provide the JS value reply to the script message sent in.
+ *
+ * Since: 2.40
+ */
+void webkit_user_script_message_reply_return_value(WebKitUserScriptMessageReply* message, JSCValue* replyValue)
+{
+    g_return_if_fail(message != nullptr);
+    g_return_if_fail(message->completionHandler);
```

Comment:
We should probably document this can be called twice

Reply:
> We should probably document this can be called twice

**CANNOT** be called more than once.